x/evm/types/msg_test.go

```diff
@@ -70,73 +73,93 @@ func (suite *MsgsTestSuite) TestMsgEthereumTx_Constructor() {
+		// {msg: "invalid recipient", to: invalidFromAddress, amount: minusOneInt, gasPrice: hundredInt, expectPass: false},
+		{msg: "nil amount - Legacy Tx", to: suite.to.Hex(), amount: nil, gasPrice: hundredInt, gasFeeCap: nil, gasTipCap: nil, expectPass: true},
+		{msg: "negative amount - Legacy Tx", to: suite.to.Hex(), amount: minusOneInt, gasPrice: hundredInt, gasFeeCap: nil, gasTipCap: nil, expectPass: false},
+		{msg: "nil gas price - Legacy Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: nil, gasFeeCap: nil, gasTipCap: nil, expectPass: false},
+		{msg: "negative gas price - Legacy Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: minusOneInt, gasFeeCap: nil, gasTipCap: nil, expectPass: false},
+		{msg: "zero gas price - Legacy Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, expectPass: true},
+		{msg: "invalid from address - Legacy Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, from: invalidFromAddress, expectPass: false},
+		{msg: "out of bound gas fee - Legacy Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: exp_2_255, gasFeeCap: nil, gasTipCap: nil, expectPass: false},
+		{msg: "nil amount - AccessListTx", to: suite.to.Hex(), amount: nil, gasPrice: hundredInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: true},
+		{msg: "negative amount - AccessListTx", to: suite.to.Hex(), amount: minusOneInt, gasPrice: hundredInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: nil, expectPass: false},
+		{msg: "nil gas price - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: nil, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: false},
+		{msg: "negative gas price - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: minusOneInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: nil, expectPass: false},
+		{msg: "zero gas price - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: true},
+		{msg: "invalid from address - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, from: invalidFromAddress, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: false},
+		{msg: "chain ID not set on AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: nil, expectPass: false},
+		{msg: "nil tx.Data - AccessList Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: false},
```

Comment:
I'd probably split this into multiple lines to ease readibility

Reply:
I agree - I kept the original format before my changes but will gladly change it.